SwiftSH/Command.swift

```diff
@@ -54,11 +54,15 @@ public class SSHCommand<T: RawLibrary>: SSHChannel<T> {
             socketSource.cancel()
         }
 
-        self.queue.async {
-            super.close()
+        self.queue.async { [weak self] in
```

Comment:
I think `self` should be retained by the block here, and in all the other places where you've added it.

Reply:
Well, that would be great but causes retain cycles and memory leaks. Here is some explanation:
http://katalisha.com/2016/01/22/ARC-Swift-closures-and-weak-self.html
http://www.thomashanning.com/retain-cycles-weak-unowned-swift/

Reply:
I'm aware that it's a retain cycle but eventually the block will be executed and the strong reference will disappear. This allow you do to something like this:
```Swift
// [...]
channel.disconnect {
    print("Disconnected")
}
channel = nil
```
With you change, `channel` would be released right away without waiting for the disconnection to complete.